sysl2/sysl/importer/swagger.go

```diff
@@ -52,3 +58,204 @@ func LoadSwaggerText(args OutputData, text string, logger *logrus.Logger) (out s
+	}
+
+	conversions := map[string]map[string]string{
+		StringTypeName: {
+			"":          StringTypeName,
+			"date":      "date",
+			"date-time": "datetime",
+			"byte":      StringTypeName,
+		},
+		"integer": {
+			"":      "int",
+			"int32": "int32",
+			"int64": "int64",
+		},
+		"number": {
+			"":       "float",
+			"double": "float",
+			"float":  "float",
+		},
+	}
+
+	if formatMap, ok := conversions[typeName]; ok {
+		if result, ok := formatMap[format]; ok {
+			return result
+		}
+		logger.Warnf("Unhandled (type, format) -> (%s, %s)\n", typeName, format)
+		return mapSwaggerTypeAndFormatToType(typeName, "", logger)
+	}
+
+	return typeName
+}
+
+func findReferencedType(schema spec.Schema, data *typeData) string {
+	if len(schema.Type) == 1 {
+		return mapSwaggerTypeAndFormatToType(schema.Type[0], schema.Format, data.logger)
+	} else if len(schema.Type) == 0 && schema.Items != nil {
+		return findReferencedType(*schema.Items.Schema, data)
+	}
+
+	if refURL := schema.Ref.GetURL(); refURL != nil {
+		return getReferenceFragment(refURL)
+	}
+
+	return ""
+}
+
+func getReferenceFragment(u *url.URL) string {
+	parts := strings.Split(u.Fragment, "/")
+	return parts[len(parts)-1]
+}
+
+type typeData struct {
+	doc        *spec.Swagger
+	knownTypes TypeList
+	logger     *logrus.Logger
+}
+
+func createTypeFromSchema(name string, schema *spec.Schema, data *typeData) Type {
+	var item Type
+	if len(schema.Properties) == 0 {
+		if isSwaggerArrayType(*schema) {
+			nested := NewStringAlias(fmt.Sprintf("%s_obj", name))
+			data.knownTypes.Add(nested)
+			item = &Array{name: name, Items: nested}
+		} else if len(schema.Enum) > 0 {
+			item = &Enum{name: name}
+		} else if refType := findReferencedType(*schema, data); refType != "" {
+			data.logger.Warnf("WARNING: swagger type '%s' is malformed\n", name)
+			t, found := data.knownTypes.Find(refType)
+			if !found {
+				if ref, ok := data.doc.Definitions[refType]; ok {
+					t = createTypeFromSchema(refType, &ref, data)
+				} else {
+					t, _ = data.knownTypes.Find(StringTypeName)
+				}
+			}
+			item = &ExternalAlias{
+				name:   name,
+				Target: t,
+			}
+		}
+	} else {
+		st := &StandardType{
+			name: name,
+		}
+		AddSwaggerProperties(st, schema.Properties, schema.Required, data)
+		item = st
+	}
+
+	data.knownTypes.Add(item)
+
+	return item
+}
+
+func (t TypeList) FindFromSchema(schema spec.Schema, data *typeData) (Type, bool) {
+	if isSwaggerArrayType(schema) {
+		items, found := t.FindFromSchema(*schema.Items.Schema, data)
+		if found {
+			return &Array{Items: items}, true
+		}
+	}
+	return t.Find(findReferencedType(schema, data))
+}
+
+func isSwaggerArrayType(schema spec.Schema) bool {
+	if len(schema.Type) == 1 {
+		typeName := schema.Type[0]
+		if typeName == "array" && schema.Items != nil {
+			return true
+		}
+	}
+
+	return false
+}
+
+func AddSwaggerProperties(s *StandardType, props map[string]spec.Schema, requiredProps []string, data *typeData) {
+	keys := []string{}
```

Comment:
Prefer nil slice over empty slice. 
https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

Reply:
k

Reply:
linter complains with the other way